### ISSUE TITLE
 TypeError: 'set' on proxy: trap returned falsish for property 'hasOwnProperty'

### ISSUE THREAD

**Issue by St-Ex-savadenn** [vuejs/vue-test-utils-next#439]
``` ts
import { shallowMount } from '@vue/test-utils'
import Button from '@/components/Button.vue'

describe('Components/Button.vue', () => {
  it('renders label when passed', () => {
    const label = 'my button'
    const wrapper = shallowMount(Button, { props: { label } })
    expect(wrapper.text()).toMatch(label)
  })
})
```

Error is

```
  ● Components/Button.vue › renders label when passed

    TypeError: 'set' on proxy: trap returned falsish for property 'hasOwnProperty'

       5 |   it('renders label when passed', () => {
       6 |     const label = 'my button'
    >  7 |     const wrapper = shallowMount(Button, { props: { label } })
         |                     ^
       8 |     expect(wrapper.text()).toMatch(label)
       9 |   })
      10 | })

      at mount (node_modules/@vue/test-utils/dist/vue-test-utils.cjs.js:2348:24)
      at shallowMount (node_modules/@vue/test-utils/dist/vue-test-utils.cjs.js:2355:12)
      at Object.<anonymous> (tests/unit/components/Button.spec.ts:7:21)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        5.006s
```

Linked to commit https://github.com/vuejs/vue-test-utils-next/commit/d05502268d64c29befa532a82b63a942b5bcbbc6

## Tests 

* 2.0.0-rc.1 : OK
* 2.0.0-rc.2 : KO
* 2.0.0-rc.3 : KO

**Comment by cexbrayat** [comment]
@St-Ex-savadenn Hmm, strange, thank you for opening an issue

Are you using the Vue 3.0.6+? If so, can you share the Button component so I can try to repro?

**Comment by St-Ex-savadenn** [comment]
I was 3.0.5 using 3.0.6 works well ! Thanks

**Comment by cexbrayat** [comment]
Happy to hear that! Yeah, the commit you linked is relying on a fix I added in vue-next, that was released only in Vue 3.0.6, explaining your issue.

**Comment by JostCrow** [comment]
This issue seems to be back with vue 3.0.7. For now I downgraded to vue 3.0.6 because that is a version that works correctly.

```
FAIL  tests/unit/example.spec.js
  Article.vue
    ✕ renders props.msg when passed (22ms)

  ● Article.vue › renders props.msg when passed

    TypeError: 'set' on proxy: trap returned falsish for property 'hasOwnProperty'

      4 | describe('Article.vue', () => {
      5 |     it('renders props.msg when passed', () => {
    > 6 |         const wrapper = shallowMount(Article, {
        |                         ^
      7 |             props: {
      8 |                 icon: "icon",
      9 |                 title: "title",

      at mount (node_modules/@vue/test-utils/dist/vue-test-utils.cjs.js:2347:24)
      at shallowMount (node_modules/@vue/test-utils/dist/vue-test-utils.cjs.js:2354:12)
      at Object.it (tests/unit/example.spec.js:6:25)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        1.176s, estimated 2s
Ran all test suites.

```
